engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -302,7 +302,7 @@ Object doMultiValue(
     if (fnAndType != null) {
       var ctx = EnsoContext.get(this);
       if (ctx.getBuiltins().any() != fnAndType.getRight()) {
-        var unwrapSelf = castTo.findTypeOrNull(fnAndType.getRight(), self, false, false);
+        var unwrapSelf = castTo.findTypeOrNull(fnAndType.getRight(), self, true, false);
```

Comment:
Reorders the _intersection types_ to make sure the value of the type we dispatch the method to is at `EnsoMultiValue.firstDispatch`.